### PR TITLE
node-forge: add pkcs5.pbkdf2 and fix optional parameter of createBuffer

### PR DESCRIPTION
**Pull request by spkjp**
Adds the definition for `pkcs5.pbkdf2` and marks the `encoding` parameter of `createBuffer` optional.

Reference:
https://github.com/digitalbazaar/forge/blob/d5bb56dc33b9d36010787c92540ed992a078f562/lib/pbkdf2.js#L38

https://github.com/digitalbazaar/forge/blob/d5bb56dc33b9d36010787c92540ed992a078f562/lib/util.js#L1405

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by westy92 — APPROVED** [review]
Looks fine, but ideally you could improve the test(s) to use this new method.

**Review by Apologiz — APPROVED** [review]

File: types/node-forge/index.d.ts [hunk 1]
```diff
@@ -516,6 +517,10 @@ declare module "node-forge" {
         function createSignedData(): PkcsSignedData;
     }
 
+    namespace pkcs5 {
+        function pbkdf2(password: string, salt: string, iterations: number, keySize: number): string;
+    }
+
     namespace md {
 
         type Encoding = "raw" | "utf8"
```
Comment (timhwang21) [inline types/node-forge/index.d.ts]: maybe we can reuse this type?

Comment (supaiku0) [reply types/node-forge/index.d.ts]: :+1:

**Comment by supaiku0** [comment]
@westy92 done.

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**